docling_core/types/experimental/document.py

```diff
@@ -0,0 +1,1106 @@
+        :param name: Optional[str]:  (Default value = None)
+        :param parent: Optional[GroupItem]:  (Default value = None)
+
+        """
+        if not parent:
+            parent = self.body
+
+        group_index = len(self.groups)
+        cref = f"#/groups/{group_index}"
+        dloc = f"{self.file_info.document_hash}{cref}"
+
+        group = GroupItem(dloc=dloc, parent=parent.get_ref())
+        if name:
+            group.name = name
+        if label:
+            group.label = label
+
+        self.groups.append(group)
+        parent.children.append(RefItem(cref=cref))
+
+        return group
+
+    def add_paragraph(
+        self,
+        label: str,
+        text: str,
+        orig: Optional[str] = None,
+        prov: Optional[ProvenanceItem] = None,
+        parent: Optional[GroupItem] = None,
+        item_cls=TextItem,
+    ):
+        """add_paragraph.
+
+        :param label: str:
+        :param text: str:
+        :param orig: Optional[str]:  (Default value = None)
+        :param prov: Optional[ProvenanceItem]:  (Default value = None)
+        :param parent: Optional[GroupItem]:  (Default value = None)
+        :param item_cls:  (Default value = TextItem)
+
+        """
+        if not parent:
+            parent = self.body
+
+        if not orig:
+            orig = text
+
+        text_index = len(self.texts)
+        cref = f"#/texts/{text_index}"
+        dloc = f"{self.file_info.document_hash}{cref}"
+        text_item = item_cls(
+            label=label,
+            text=text,
+            orig=orig,
+            dloc=dloc,
+            parent=parent.get_ref(),
+        )
+        if prov:
+            text_item.prov.append(prov)
+
+        self.texts.append(text_item)
+        parent.children.append(RefItem(cref=cref))
+
+        return text_item
+
+    def add_table(
+        self,
+        data: BaseTableData,
+        caption: Optional[Union[TextItem, RefItem]] = None,  # This is not cool yet.
+        prov: Optional[ProvenanceItem] = None,
+        parent: Optional[GroupItem] = None,
+    ):
+        """add_table.
+
+        :param data: BaseTableData:
+        :param caption: Optional[Union[TextItem:
+        :param RefItem]]:  (Default value = None)
+        :param # This is not cool yet.prov: Optional[ProvenanceItem]
+        :param parent: Optional[GroupItem]:  (Default value = None)
+
+        """
+        if not parent:
+            parent = self.body
+
+        table_index = len(self.tables)
+        cref = f"#/tables/{table_index}"
+        dloc = f"{self.file_info.document_hash}{cref}"
+
+        tbl_item = TableItem(
+            label=DocItemLabel.TABLE, data=data, dloc=dloc, parent=parent.get_ref()
+        )
+        if prov:
+            tbl_item.prov.append(prov)
+        if caption:
+            tbl_item.captions.append(caption.get_ref())
+
+        self.tables.append(tbl_item)
+        parent.children.append(RefItem(cref=cref))
+
+        return tbl_item
+
+    def add_figure(
```

Comment:
We could avoid some duplicate code among `add_figure`, `add_table`, and `add_paragraph`